filament/src/ColorGrading.cpp

```diff
@@ -104,20 +117,24 @@ FColorGrading::FColorGrading(FEngine& engine, const Builder& builder) {
             for (size_t g = 0; g < LUT_DIMENSION; g++) {
                 for (size_t r = 0; r < LUT_DIMENSION; r++) {
                     float3 v = float3{ r, g, b } * (1.0f / (LUT_DIMENSION - 1u));
+
                     // LogC encoding
-                    v.x = lutToLinear(v.x);
-                    v.y = lutToLinear(v.y);
-                    v.z = lutToLinear(v.z);
-                    // ACES tone mapping
+                    v = lutToLinear(v);
+
                     v = toneMapper(v);
-                    // sRGB encoding
+
+                    // TODO: allow to customize the output color space
                     v = image::linearTosRGB(v);
+
                     *p++ = half4{ v, 0 };
                 }
             }
         });
         js.run(job);
     }
+
+    // TODO: Should we do a runAndRetain() and defer the wait() + texture creation until
+    //       getHwHandle() is invoked?
```

Comment:
That's an interesting idea, but we've never really used the JobSystem in that way.
Also if we did this we'd have to handle the case of multiple calls in a row. 
As long as we're talking < 10ms I wouldn't go there.

Reply:
Yeah that's why I just left it as a TODO. Handling multiple calls in a row wouldn't be difficult I think but it does complicate things a bit.